model/host/hostutil.go

```diff
@@ -992,6 +997,37 @@ func (h *Host) withTaggedProcs(ctx context.Context, env evergreen.Environment, t
 	return handleTaggedProcs(procs)
 }
 
+func (h *Host) CheckTaskDataFetched(ctx context.Context, env evergreen.Environment) error {
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+
+	return h.withTaggedProcs(ctx, env, evergreen.HostFetchTag, func(procs []jasper.Process) error {
+		grip.WarningWhen(len(procs) > 1, message.Fields{
+			"message":   "host is attempting to fetch task data multiple times",
+			"num_procs": len(procs),
+			"host_id":   h.Id,
+			"distro":    h.Distro.Id,
+		})
+		catcher := grip.NewBasicCatcher()
+		for _, proc := range procs {
+			err := util.Retry(
+				ctx,
+				func() (bool, error) {
+					if proc.Complete(ctx) {
+						return false, nil
+					}
+					return true, errors.New("fetching task data not finished")
+				}, 1000, time.Second, 15*time.Second)
```

Comment:
I think it's a concern to have long-running jobs (i.e. blocking this job until the task is fully fetched) because usually jobs should be short-lived with some retry logic to re-enqueue instead of waiting. Can we do fewer retry attempts? If it fails, we can just just add an error to the job + return so that `tryRequeue()` will retry this check later on a new job.

Reply:
Sure, that's a reasonable argument